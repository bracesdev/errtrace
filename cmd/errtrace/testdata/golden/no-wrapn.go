//go:build ignore

// @runIf options=no-wrapn
package foo

import "example.com/bar"

func hasTwo() (int, error) {
	// Same names as used by rewriting, with different types to verify scoping.
	r1 := true
	r2 := false
	return bar.Two()
}

func hasThree() (string, int, error) {
	return bar.Three()
}

func hasFour() (string, int, bool, error) {
	return bar.Four()
}

func hasFive() (a int, b bool, c string, d int, e error) {
	return bar.Five()
}

func hasSix() (a int, b bool, c string, d int, e bool, f error) {
	return bar.Six()
}

func hasSeven() (a int, b bool, c string, d int, e bool, f string, g error) {
	return bar.Seven()
}

func nonFinalError() (error, bool) {
	return bar.NonFinalError() // want:"skipping function with non-final error return"
}

func multipleErrors() (x int, err1, err2 error) {
	return bar.MultipleErrors() // want:"skipping function with multiple error returns"
}

func invalid() (x int, err error) {
	return 42 // want:"skipping function with incorrect number of return values: got 1, want 2"
}

func nestedExpressions() (int, error) {
	return func() (int, error) {
		r1 := true
		r2 := false
		return bar.Two()
	}()
}
