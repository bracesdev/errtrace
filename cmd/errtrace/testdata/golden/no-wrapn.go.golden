//go:build ignore

// @runIf options=no-wrapn
package foo

import "example.com/bar"; import "braces.dev/errtrace"

func hasTwo() (int, error) {
	// Same names as used by rewriting, with different types to verify scoping.
	r1 := true
	r2 := false
	{ r1, r2 := bar.Two(); return r1, errtrace.Wrap(r2) }
}

func hasThree() (string, int, error) {
	{ r1, r2, r3 := bar.Three(); return r1, r2, errtrace.Wrap(r3) }
}

func hasFour() (string, int, bool, error) {
	{ r1, r2, r3, r4 := bar.Four(); return r1, r2, r3, errtrace.Wrap(r4) }
}

func hasFive() (a int, b bool, c string, d int, e error) {
	{ r1, r2, r3, r4, r5 := bar.Five(); return r1, r2, r3, r4, errtrace.Wrap(r5) }
}

func hasSix() (a int, b bool, c string, d int, e bool, f error) {
	{ r1, r2, r3, r4, r5, r6 := bar.Six(); return r1, r2, r3, r4, r5, errtrace.Wrap(r6) }
}

func hasSeven() (a int, b bool, c string, d int, e bool, f string, g error) {
	{ r1, r2, r3, r4, r5, r6, r7 := bar.Seven(); return r1, r2, r3, r4, r5, r6, errtrace.Wrap(r7) }
}

func nonFinalError() (error, bool) {
	return bar.NonFinalError() // want:"skipping function with non-final error return"
}

func multipleErrors() (x int, err1, err2 error) {
	return bar.MultipleErrors() // want:"skipping function with multiple error returns"
}

func invalid() (x int, err error) {
	return 42 // want:"skipping function with incorrect number of return values: got 1, want 2"
}

func nestedExpressions() (int, error) {
	{ r1, r2 := func() (int, error) {
		r1 := true
		r2 := false
		{ r1, r2 := bar.Two(); return r1, errtrace.Wrap(r2) }
	}(); return r1, errtrace.Wrap(r2) }
}
